config/puma.rb

```diff
@@ -42,6 +42,9 @@
 # Specifies the `pidfile` that Puma will use.
 pidfile ENV.fetch("PIDFILE") { "tmp/pids/server.pid" }
 
+# Optionally enable the Puma Control/Metrics API
+activate_control_app "tcp://127.0.0.1:9293", { auth_token: ENV.fetch("PUMA_CONTROL_AUTH_TOKEN") } if ENV["PUMA_CONTROL_ENABLED"].present?
```

Comment:
Two questions.

1. Are 2 variables needed? Isn't it possible to enable when token is present?
2. Is token even needed? Is the Metrics API exposed to the public?

Reply:
I will answer your questions one-by-one. 

> Are 2 variables needed? Isn't it possible to enable when token is present?

I need to coordinate the release of this for testing and ensuring the integration between Datadog and puma is reliable. Seeing we do not having branch environments, I can only do this with feature flags/environment variables for to enable for staging and production. 

Maybe after this PR has been merged we can reduce the flags down into something simpler.

> Is token even needed? Is the Metrics API exposed to the public?

The control server also lets you shutdown, restart and manage other process related actions for puma. The port will be exposed within Kubernetes, so i want to ensure the endpoint is well protected from any potential threats.